docs/rules/optimal-quantifier-concatenation.md

```diff
@@ -43,7 +43,36 @@ var foo = /\w+(?:(a)|b)*/;
 
 ## :wrench: Options
 
-Nothing.
+```json5
+{
+  "regexp/optimal-quantifier-concatenation": [
+    "error",
+    {
+        "capturingGroups": "report"
+    }
+  ]
+}
+```
+
+### `capturingGroups`
+
+The type of concatenation this rule reports might be intentional around capturing groups. This option allows you turn of false unfixable reports around capturing groups.
+
+- `capturingGroups: "report"` (_default_)
+
+  Concatenations around quantifiers will be reported.
+
+- `capturingGroups: "ignore"`
```

Comment:
The `"ignore"` option also suppresses the next RegExp warning. Is this intentional?

```js
/\d+(\d*)/u
```

It doesn't make sense because the capture group always returns blanks.

Reply:
> It doesn't make sense because the capture group always returns blanks.

Yes, but finding nonsensical capturing groups is not the purpose of this rule. This rule just also finds some bad capturing groups from time to time.

We should add a new rule for finding bad capturing groups.